mono/metadata/monitor.c

```diff
@@ -1438,3 +1438,15 @@ ves_icall_System_Threading_Monitor_Monitor_Enter (MonoObject *obj)
 {
 	mono_monitor_enter_internal (obj);
 }
+
+#if ENABLE_NETCORE
```

Comment:
nit: `#ifdef` (we have a mix of `#if ENABLE_NETCORE` and `#ifdef ENABLE_NETCORE` makes sense to unify it

Reply:
Yeah, I wanted to do that... but then I reverted the whole experiment and lost this small change. :)

I noticed that there's a mix already. Unless I will need to amend the PR due to some other change I'd prefer to do the `#if`/`#ifdef` unification in separate PR once there is agreement on the preferred way.

Reply:
@jaykrell I don't disagree, in general. In this specific instance it is really `#if` vs `#ifdef` though. The netcore version is significantly different from the regular runtime. It is not just an optional feature where we can easily rely on compiler dead code elimination and regular `if`.